src/core/Basics/Sample.cpp

```diff
@@ -51,18 +50,18 @@ static RubberBand::RubberBandStretcher::Options compute_rubberband_options( cons
 
 
 /* EnvelopePoint */
-EnvelopePoint::EnvelopePoint() : Object( EnvelopePoint::__class_name ), frame( 0 ), value( 0 ) 
```

Comment:
How come you removed the `H2Core::Object` in here? It's used for debugging and shouldn't add too much weight.

Reply:
> How come you removed the `H2Core::Object` in here? It's used for debugging and shouldn't add too much weight.

Hum, I was maybe a bit to quick with this. I think EnvelopePoint should go away in favor of a proper Envelope class.
Besides, I have a pending patch which removes the H2Core::Object's overhead... 

My point was to get rid of the unique_ptr around EnvelopePoint in envelope, so I can access them. I don't remember the details right now, but pybind11 caused me some troubles with those std::shared_ptr<std::unique_ptr<EnvelopePoint>> ...

ihmo, EnvelopePoints should be plain dummy structs with copy semantics, and let the (to be written) Envelope let deal with them.

Reply:
Hi @theGreatWhiteShark I just restored EnvelopePoint inheritance of H2Core::Object.
Regards

Reply:
> I think EnvelopePoint should go away in favor of a proper Envelope class.
Besides, I have a pending patch which removes the H2Core::Object's overhead...

Could you split off these changes and make a separate PR? Such fundamental changes in the code base require the consent of at least a majority of all maintainers. This will happen much faster if it's a concise one.

One thing to think about in advance (and I haven't done yet, so, there is a chance none of this makes sense): The envelope in the sample editor is quite similar to the automation path. Maybe there could be a common basis.

Reply:
> > I think EnvelopePoint should go away in favor of a proper Envelope class.
> > Besides, I have a pending patch which removes the H2Core::Object's overhead...
> 
> Could you split off these changes and make a separate PR? Such fundamental changes in the code base require the consent of at least a majority of all maintainers. This will happen much faster if it's a concise one.
> 

Yes, I will do a separate PR on this.

> One thing to think about in advance (and I haven't done yet, so, there is a chance none of this makes sense): The envelope in the sample editor is quite similar to the automation path. Maybe there could be a common basis.

I will have a look at those automation path. If we could factor out some code, it would be nice.

Reply:
Please see #1318